### PR TITLE
RSDK-7515 update ptg inputs to no longer be negative

### PR DESCRIPTION
**Pull request by biotinker**
This change should be a transparent no-op from a user's perspective. Prerequisite to the testing updates coming with RSDK-7558

### CODE REVIEW

File: motionplan/tpSpaceRRT.go [hunk 1]
```diff
@@ -1054,8 +1054,8 @@ func extractTPspacePath(startMap, goalMap map[node]node, pair *nodePair) []node
 			q: []referenceframe.Input{
 				goalReached.Q()[0],
 				goalReached.Q()[1],
+				goalReached.Q()[3],
```
Comment (raybjork) [inline motionplan/tpSpaceRRT.go:1057]: Why are these out of order?

**Review by raybjork — CHANGES_REQUESTED** [review]

File: motionplan/tpSpaceRRT.go (hunk 1 above)
Comment (biotinker) [reply motionplan/tpSpaceRRT.go:1057]: The tpspace algorithm calculates the goal tree in reverse. This is where those inputs are rectified to be with regard to the forwards tree.

File: motionplan/tpspace/ptg.go [hunk 2]
```diff
@@ -154,11 +154,11 @@ func invertComputedPTG(forwardsPTG []*TrajNode) []*TrajNode {
 		)
 		flippedTraj = append(flippedTraj,
 			&TrajNode{
-				flippedPose,
-				startNode.Dist - fwdNode.Dist,
-				startNode.Alpha,
-				fwdNode.LinVel,
-				fwdNode.AngVel * -1,
+				Pose:   flippedPose,
+				Dist:   fwdNode.Dist,
```
[the remark this answers is not in the document]
Comment (biotinker) [reply motionplan/tpspace/ptg.go:158]: The subtraction needs to be removed for `CurrentInputs` to work properly. We need to have the actual node dists passed through in order for poses to be computed properly down the line. e.g. if we are going from dist 150 to 50, then we want the nodes to be in that order.

Prior to this change the inputs would have been reported as going from -50 to -150, which needed to be handled very specially in order to have the dists come out properly.

This all follows on to the prior change that made ptgs 4dof and allowed partial addressing of trajectories, which really did not play nicely with the invertible ptgs. This PR finally fixes the weird math special-cases for negative distances by simply providing a start and end point.

File: motionplan/tpspace/ptgGroupFrame.go [hunk 3]
```diff
@@ -199,24 +201,18 @@ func (pf *ptgGroupFrame) Transform(inputs []referenceframe.Input) (spatialmath.P
...
 // The above is inverted with negative distances, requiring some complicated logic which should be radically simplified by RSDK-7515.
 func (pf *ptgGroupFrame) Interpolate(from, to []referenceframe.Input, by float64) ([]referenceframe.Input, error) {
-	if len(from) != len(pf.DoF()) {
-		return nil, referenceframe.NewIncorrectInputLengthError(len(from), len(pf.DoF()))
+	if err := pf.validInputs(from); err != nil {
+		return nil, err
 	}
-	if len(to) != len(pf.DoF()) {
-		return nil, referenceframe.NewIncorrectInputLengthError(len(to), len(pf.DoF()))
+	if err := pf.validInputs(to); err != nil {
```
[the remark this answers is not in the document]
Comment (biotinker) [reply motionplan/tpspace/ptgGroupFrame.go:207]: Yes!

**Review by raybjork — APPROVED** [review]

**Pull request merged**